.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,92 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the
+// README at: https://github.com/devcontainers/templates/tree/main/src/docker-existing-dockerfile
+{
+  "name": "Codespace setup",
+  "image": "mcr.microsoft.com/devcontainers/base:ubuntu",
+
+  // For details, see https://containers.dev/implementors/json_reference/#min-host-reqs
+  "hostRequirements": {
+    "memory": "6gb"
+  },
+
+  // Features to add to the dev container. More info: https://containers.dev/features.
+  "features": {
+    // Enable running docker in docker
+    "ghcr.io/devcontainers/features/docker-in-docker:2": {},
+    "ghcr.io/devcontainers/features/sshd:1": {},
+    "ghcr.io/devcontainers/features/github-cli:1": {},
+    "ghcr.io/devcontainers/features/java:1.4.1": {
+      "version": "none",
+      "installMaven": "true",
+      "installGradle": "false"
+    },
+    "ghcr.io/devcontainers-contrib/features/ant-sdkman:2": {}
+  },
+
+  "customizations": {
+    "vscode": {
+      "settings": {
+        "files.watcherExclude": {
+          "**/target": true,
+          "**/.bloop": true,
+          "**/node_modules": true
+        },
+        "search.exclude": {
+          "*/target": true,
+          "**/.bloop": true,
+          "**/node_modules": true
+        },
+        "java.configuration.updateBuildConfiguration": "automatic",
+        "java.compile.nullAnalysis.mode": "disabled"
+      },
+      "extensions": [
```

Comment:
can devs at their own extensions? Is this just the extensions it starts with?

Reply:
Yeah this is just what it starts with. Devs can add their own extensions, but ideally anything useful would be added here so everyone can benefit from it.

Reply:
If you add your own you probably have to re-add them if the environment dies :/

Reply:
Yeah that's true too!